File: app/src/test/java/com/example/seki/trainingRecyclerView/DataTest.java
package com.example.seki.trainingRecyclerView;

import org.junit.Test;

import static org.hamcrest.Matchers.is;
import static org.junit.Assert.*;

public class DataTest {

    private static final String TITLE = "test";
    private static final String DESCRIPTION = "this is test data";

    @Test
    public void dataのコンストラクタのテスト() {
        Data data = new Data(TITLE, DESCRIPTION);
        assertThat(data.getTitle().get(), is(TITLE));
        assertThat(data.getDescription().get(), is(DESCRIPTION));
        assertThat(data.getClickCount().get(), is("0"));
    }

    @Test
    public void クリックカウントが変更されること() {
        Data data = new Data(TITLE, DESCRIPTION);
        data.getClickCount().set("1");
        assertThat(data.getClickCount().get(), is("1"));
    }
}

File: app/src/test/java/com/example/seki/training_recycler_view/DataTest.kt
package com.example.seki.training_recycler_view

import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.assertThat
import org.junit.Test

class DataTest {

    companion object {
        const val TITLE = "title"
        const val DESCRIPTION = "description"
    }

    @Test fun `Dataのインスタンスが作成できること`() {
        val data = Data(TITLE, DESCRIPTION)

        assertThat(data.title.get(), `is`(TITLE))
        assertThat(data.description.get(), `is`(DESCRIPTION))
        assertThat(data.clickCount.get(), `is`(0.toString()))
    }
}